### ISSUE TITLE
How to properly upload font .woff2 files to a style css note?

### ISSUE THREAD

**Issue by PAK90** [zadam/trilium#1098]
Using 0.42.7; trying to figure this out. The example Steel Blue style has two children files, both .woff2 font files. However, they don't seem to be referenced in the actual Steel Blue .css file; instead it points to some `/custom/fonts/raleway.woff2` url that I can't find. I'm trying to make my own style, so I obtained a .woff2 file, and copied Steel Blue to start modifying it. I uploaded the .woff2 to my new .css file (so now it's a child, just like Steel Blue) and changed the `src` attribute of the first font-face block to `url('ibm-plex-sans-v7-latin-300.woff2') format ('woff2');`. I also changed the `font-family` attribute to 'IBM Plex Sans' of course, but that doesn't help.

This gives me the following error in the console: `GET http://127.0.0.1:37840/api/notes/download/ibm-plex-sans-v7-latin-300.woff2 net::ERR_ABORTED 404 (Not Found)`, and it falls back to the default serif font. What am I missing here? I also tried prefixing the font url with `/custom/fonts` as in the example, but that doesn't work either.

**Comment by zadam** [comment]
Hi, the missing piece is that the fonts are exposed using "custom resource provider" as described here: https://github.com/zadam/trilium/wiki/Custom-request-handler#custom-resource-provider

**Comment by zDEFz** [comment]
Would it be possible to make an example?
I'm having the same understanding problem.

So I've used this... (anonymized)
`
Font.css, #appTheme #customResourceProvider=myFont.woff2 
@font-face {
  font-family: 'myFont';
  font-style: normal;
  font-weight: 400;
  src: url('/custom/fonts/myFont.woff2') format('woff2');
}
`
child is:

myFont.woff2

Now making sure my appTheme **Font** is selected, and restart application.

Nothing has changed. 

Is this me, being unable to decipher the instructions on the provided link?

**Comment by zDEFz** [comment]
I found out how to do it and post a example once my Theme is done...

**Comment by zDEFz** [comment]
@PAK90 the Fonts need to wear the "#customRessourceProvider" tag, without any argument.

Afterwards, they can be acessed from, defining in css, via Font-Face.
I got the info after digging on https://github.com/raphwriter/trilium-theme-melon

@zadam, it was very difficult to understand, the way it's written in the Wiki. 
The Wiki is good to have, but some parts are not easy to consume. This is a todo.
I tried myself and added some details. https://github.com/zadam/trilium/wiki/Custom-request-handler
